tests/functional/src/main/java/org/mule/functional/api/classloading/isolation/AetherClassPathClassifier.java

```diff
@@ -0,0 +1,692 @@
+   */
+  private PluginUrlClassification buildPluginUrlClassification(Artifact pluginArtifact, ClassPathClassifierContext context,
+                                                               File baseResourcesFolder,
+                                                               ExtensionsTestInfrastructureDiscoverer extensionsInfrastructure) {
+    List<Dependency> managedDependencies = dependencyResolver.readArtifactDescriptor(pluginArtifact).getManagedDependencies();
+
+    List<URL> urls = toUrl(dependencyResolver
+        .resolveDependencies(
+                             new Dependency(pluginArtifact,
+                                            COMPILE),
+                             Collections.<Dependency>emptyList(),
+                             managedDependencies,
+                             orFilter(classpathFilter(COMPILE),
+                                      new PatternExclusionsDependencyFilter(context.getExcludedArtifacts()))));
+
+    if (context.isExtensionMetadataGenerationEnabled()) {
+      URL generatedTestResources =
+          buildExtensionPluginMetadata(baseResourcesFolder, extensionsInfrastructure, pluginArtifact, urls);
+
+      if (generatedTestResources != null) {
+        List<URL> appendedTestResources = newArrayList(generatedTestResources);
+        appendedTestResources.addAll(urls);
+        urls = appendedTestResources;
+      }
+    }
+    // TODO (gfernandes): How could I check if exported classes belong to this plugin?
+    return new PluginUrlClassification(toId(pluginArtifact), urls, newArrayList(context.getExportPluginClasses()));
+  }
+
+  /**
+   * @return an {@link ExtensionManagerAdapter} that would be used to register the extensions, later it would be discarded.
+   */
+  private ExtensionManagerAdapter createExtensionManager() {
+    DefaultExtensionManager extensionManager = new DefaultExtensionManager();
+    extensionManager.setMuleContext(new DefaultMuleContext() {
+
+      @Override
+      public MuleRegistry getRegistry() {
+        return new MuleRegistryHelper(new DefaultRegistryBroker(this), this);
+      }
+    });
+    try {
+      extensionManager.initialise();
+    } catch (InitialisationException e) {
+      throw new RuntimeException("Error while initialising the extension manager", e);
+    }
+    return extensionManager;
+  }
+
+  /**
+   * Generates the extension metadata if the plugin has a class annotated with the {@link Extension} annotation.
+   *
+   * @param baseResourcesFolder {@link File} base folder to write metadata's file
+   * @param extensionsInfrastructure {@link ExtensionsTestInfrastructureDiscoverer} to generate metadata
+   * @param pluginArtifact {@link Artifact} representing the plugin
+   * @param urls plugin {@link URL}'s resolved
+   * @return {@link URL} to the folder where the metadata was created or null if the plugin is not an extension
+   */
+  private URL buildExtensionPluginMetadata(File baseResourcesFolder,
```

Comment:
Can you extract this code into a new class in order to make the code more testeable?